### DISCUSSION TITLE
Failed to compile with CPC interface with latest release.

### DISCUSSION THREAD

**Discussion by amit3barthwal** [openthread/openthread#9169] [Q&A] [answered]
# Details
We are recently trying to get more information on the thread network side. I have check the api **"meshdiag"** is available on the latest release openthread CLI. But when i tried to compile the openthread code i am not able to compile with the CPC interface.

# Build Command

 cmake -B build_1_3 -G Ninja -DOT_MULTIPAN_RCP=1 -DOT_RCP_RESTORATION_MAX_COUNT=10 -DOTBR_BACKBONE_ROUTER=TRUE -DOT_MAX_POWER_TABLE=0 -DOT_CHANNEL_MANAGER=0 -DOT_CHANNEL_MONITOR=0 -DOT_THREAD_VERSION=1.3 **-DOT_POSIX_CONFIG_RCP_BUS=CPC** -DBUILD_TESTING=FALSE -DCMAKE_BUILD_TYPE=Release -DOTBR_DBUS=TRUE -DOTBR_REST=TRUE -DOTBR_MDNS=avahi -DOTBR_SRP_ADVERTISING_PROXY=TRUE -DOTBR_VENDOR_NAME="SmartThings" -DOTBR_PRODUCT_NAME="WCH" -DCMAKE_TOOLCHAIN_FILE=
 
 
# Error Logs:-

 third_party/openthread/repo/src/posix/platform/CMakeFiles/openthread-posix.dir/radio.cpp.o
/usr/bin/c++ -DOPENTHREAD_CONFIG_BACKBONE_ROUTER_DUA_NDPROXYING_ENABLE=0 -DOPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE=1 -DOPENTHREAD_CONFIG_BORDER_AGENT_ENABLE=1 -DOPENTHREAD_CONFIG_BORDER_AGENT_ID_ENABLE=1 -DOPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE=1 -DOPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE=0 -DOPENTHREAD_CONFIG_CHANNEL_MANAGER_ENABLE=0 -DOPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE=0 -DOPENTHREAD_CONFIG_COAP_API_ENABLE=1 -DOPENTHREAD_CONFIG_COAP_SECURE_API_ENABLE=1 -DOPENTHREAD_CONFIG_COMMISSIONER_ENABLE=1 -DOPENTHREAD_CONFIG_DATASET_UPDATER_ENABLE=1 -DOPENTHREAD_CONFIG_DNSSD_SERVER_ENABLE=0 -DOPENTHREAD_CONFIG_DNS_CLIENT_ENABLE=1 -DOPENTHREAD_CONFIG_DNS_UPSTREAM_QUERY_ENABLE=0 -DOPENTHREAD_CONFIG_ECDSA_ENABLE=1 -DOPENTHREAD_CONFIG_ENABLE_BUILTIN_MBEDTLS=1 -DOPENTHREAD_CONFIG_ENABLE_BUILTIN_MBEDTLS_MANAGEMENT=0 -DOPENTHREAD_CONFIG_FILE=\"openthread-core-posix-config.h\" -DOPENTHREAD_CONFIG_HEAP_EXTERNAL_ENABLE=1 -DOPENTHREAD_CONFIG_HISTORY_TRACKER_ENABLE=1 -DOPENTHREAD_CONFIG_IP6_BR_COUNTERS_ENABLE=1 -DOPENTHREAD_CONFIG_IP6_SLAAC_ENABLE=1 -DOPENTHREAD_CONFIG_JOINER_ENABLE=1 -DOPENTHREAD_CONFIG_LOG_CLI=1 -DOPENTHREAD_CONFIG_LOG_LEVEL=OT_LOG_LEVEL_INFO -DOPENTHREAD_CONFIG_LOG_LEVEL_DYNAMIC_ENABLE=1 -DOPENTHREAD_CONFIG_MAC_FILTER_ENABLE=1 -DOPENTHREAD_CONFIG_MAX_STATECHANGE_HANDLERS=3 -DOPENTHREAD_CONFIG_MLE_LINK_METRICS_SUBJECT_ENABLE=1 -DOPENTHREAD_CONFIG_MLE_STEERING_DATA_SET_OOB_ENABLE=1 -DOPENTHREAD_CONFIG_MLR_ENABLE=1 -DOPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE=1 -DOPENTHREAD_CONFIG_NAT64_BORDER_ROUTING_ENABLE=0 -DOPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE=0 -DOPENTHREAD_CONFIG_NCP_HDLC_ENABLE=1 -DOPENTHREAD_CONFIG_NETDATA_PUBLISHER_ENABLE=1 -DOPENTHREAD_CONFIG_PING_SENDER_ENABLE=1 -DOPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE=1 -DOPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE=1 -DOPENTHREAD_CONFIG_POSIX_SETTINGS_PATH=\"/var/lib/thread\" -DOPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE=1 -DOPENTHREAD_CONFIG_RADIO_LINK_TREL_ENABLE=0 -DOPENTHREAD_CONFIG_SRP_CLIENT_ENABLE=1 -DOPENTHREAD_CONFIG_SRP_SERVER_ENABLE=1 -DOPENTHREAD_CONFIG_TCP_ENABLE=0 -DOPENTHREAD_CONFIG_THREAD_VERSION=OT_THREAD_VERSION_1_3 -DOPENTHREAD_CONFIG_TMF_ANYCAST_LOCATOR_ENABLE=1 -DOPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE=1 -DOPENTHREAD_CONFIG_TMF_NETDIAG_CLIENT_ENABLE=1 -DOPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE=1 -DOPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE=1 -DOPENTHREAD_CONFIG_UDP_FORWARD_ENABLE=0 -DOPENTHREAD_CONFIG_UPTIME_ENABLE=1 -DOPENTHREAD_FTD=1 -DOPENTHREAD_MTD=0 -DOPENTHREAD_PLATFORM_POSIX=1 -DOPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE=1 

/util/third_party/ot-br-posix/third_party/openthread/repo/src/posix/platform/radio.cpp
**/util/third_party/ot-br-posix/third_party/openthread/repo/src/posix/platform/radio.cpp:64:2: error: #error** "OPENTHREAD_POSIX_CONFIG_RCP_BUS only allows OT_POSIX_RCP_BUS_UART, OT_POSIX_RCP_BUS_SPI and " "OT_POSIX_RCP_BUS_VENDOR!"
   64 | #error "OPENTHREAD_POSIX_CONFIG_RCP_BUS only allows OT_POSIX_RCP_BUS_UART, OT_POSIX_RCP_BUS_SPI and " \
      |  ^~~~~

# Analysis

When compiling with **"-DOT_POSIX_CONFIG_RCP_BUS=CPC"** option, i am getting above erroe. 
If i remove this option, then compiling successfully.

**Could you please help me to know, how can i enable **"CPC"** interface option as it is available in the previous opthread version. i.e 20200818 release.**

Current working version 20200818 release working with CPC interface.
Latest release not working with CPC interface option.

**Comment by jwhui** [comment]
I believe CPC is a technology specific to the Silicon Labs platform/SDK and not a part OpenThread git repo.

**Comment by amit3barthwal** [comment]
I am able to compile the same in the previous version of openthread. 20200818 release

**Reply by jwhui** [reply, marked as answer]
As mentioned in https://github.com/openthread/openthread/discussions/9169#discussioncomment-6158704, CPC is not defined anywhere in the `thread-reference-20200818` tag.

Also, see the relevant code below:

https://github.com/openthread/openthread/blob/54b31928cf65803a9cd83c9f97061b64e465aaf7/src/posix/platform/radio.cpp#L38-L52

**Reply by amit3barthwal** [reply]
Okay.. I will check changes on the latest SDK. Thanks for sharing information.

**Comment by romacdon** [comment]
@amit3barthwal In the 4.3.0 release of the Silabs GSDK we have modified how CPC is integrated with OpenThread.  In this release we are utilizing the VENDOR hooks provided by openthread.  Please see https://www.silabs.com/documents/public/application-notes/an1333-concurrent-protocols-with-802-15-4-rcp.pdf for details on building with CPC with the latest GSDK.  Also, if you continue to have issues please feel free to open a salesforce ticket with Silicon Labs as CPC is not part of the openthread project.
